dotCMS/src/main/java/com/dotcms/variant/VariantAPIImpl.java

```diff
@@ -105,6 +109,55 @@ public void delete(String id) throws DotDataException {
         Logger.debug(this, ()-> "Deleting Variant: " + variant);
 
         variantFactory.delete(id);
+
+        deleteAllContentletVersions(variant);
+        deleteMultiTrees(variant);
+
+    }
+
+    private void deleteMultiTrees(final Variant variant) throws DotDataException {
+        final MultiTreeAPI multiTreeAPI = APILocator.getMultiTreeAPI();
+        final List<MultiTree> multiTrees = multiTreeAPI.getMultiTrees(variant);
+        multiTreeAPI.deleteMultiTree(multiTrees);
+
+    }
+
+    private static void deleteAllContentletVersions(Variant variant) throws DotDataException {
+        try {
+            APILocator.getContentletAPI().getAllContentByVariants(APILocator.systemUser(), false , variant.name())
+                    .stream()
+                    .map(contentlet -> getIdentifier(contentlet))
+                    .flatMap(identifier -> getVersions(variant, identifier).stream())
+                    .forEach(contentlet -> deleteContentlet(contentlet));
+        } catch (DotSecurityException e) {
```

Comment:
How can we catch a `DotSecurityException` from whatever is inside the `.forEach()` consumer?

Reply:
this method APILocator.getContentletAPI().getAllContentByVariants can throw a DotSecurityException outside of the forEach